drivers/netdev2_ieee802154/netdev2_ieee802154.c

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (C) 2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <string.h>
+#include <stdbool.h>
+
+#include "net/eui64.h"
+#include "net/ieee802154.h"
+#include "net/netdev2.h"
+
+#include "net/netdev2_ieee802154.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static int _get_iid(netdev2_ieee802154_t *dev, eui64_t *value, size_t max_len)
+{
+    uint8_t *addr;
+    uint16_t addr_len;
+
+    if (max_len < sizeof(eui64_t)) {
+        return -EOVERFLOW;
+    }
+
+    if (dev->flags & NETDEV2_IEEE802154_SRC_MODE_LONG) {
+        addr_len = IEEE802154_LONG_ADDRESS_LEN;
+        addr = dev->long_addr;
+    }
+    else {
+        addr_len = IEEE802154_SHORT_ADDRESS_LEN;
+        addr = dev->short_addr;
+    }
+    ieee802154_get_iid(value, addr, addr_len);
+
+    return sizeof(eui64_t);
+}
+
+int netdev2_ieee802154_get(netdev2_ieee802154_t *dev, netopt_t opt, void *value,
+                           size_t max_len)
+{
+    int res = -ENOTSUP;
+
+    switch (opt) {
+        case NETOPT_ADDRESS:
+            if (max_len < sizeof(dev->short_addr)) {
```

Comment:
Hm. I remember we somehow decided that all those parameter checks bloat a lot, and a simple assert() is enough.